src/nbl/video/CVulkanCommandPool.cpp

```diff
@@ -12,6 +12,18 @@ CVulkanCommandPool::~CVulkanCommandPool()
     vk->vk.vkDestroyCommandPool(vulkanDevice->getInternalObject(), m_vkCommandPool, nullptr);
 }
 
+void CVulkanCommandPool::reset()
+{
+	// Free everything in the memory pool from all command buffers
+	// TODO figure out if this is the best way to do it
+	mempool.reset();
+
+	const CVulkanLogicalDevice* vulkanDevice = static_cast<const CVulkanLogicalDevice*>(getOriginDevice());
+	auto* vk = vulkanDevice->getFunctionTable();
+	vk->vk.vkResetCommandPool(vulkanDevice->getInternalObject(), m_vkCommandPool, 0);
+	IGPUCommandPool::reset();
```

Comment:
you want to increment the counter at the start

Reply:
Resolved.